### DISCUSSION TITLE
Detect click on column header and display a different context menu.

### DISCUSSION THREAD

**Discussion by ajboni** [nick-keller/react-datasheet-grid#142] [Feature requests]
Hello! Great library!!
I was wondering if there is any possibility to get if we clicked on a column header and display a different context menu.
I'm using this as starting point
https://github.com/Equify/react-datasheet-grid/blob/master/src/components/ContextMenu.tsx

the thing is that the whole column is selected if right-clicking on a header so active cell is never null...
 https://react-datasheet-grid.netlify.app/docs/api-reference/ref

```  return (
    <DynamicDataSheetGrid
      ref={gridRef}
      value={data}
      onChange={setData}
      columns={columnsMemo}
      createRow={createRow}
      contextMenuComponent={(e) => GridContextMenu(e, gridRef.current)}
    />
  );
```

**Comment by nick-keller** [comment]
Hey @ajboni ,
Could you please explain your use case in detail? I'm trying to figure out what exactly would be needed.
Thx!

**Reply by ajboni** [reply]
Hey @nick-keller , 
I'm trying to give the user an option to right click the column header and have a contextual menu to edit the title, remove it, add a column before/after, change the column type. Seems to be more appropriate than display it on every right click on every cell.

Hope it makes sense! Thanks.

**Comment by nick-keller** [comment]
Hey @ajboni, this is now available in 4.2.0 ;)

The context menu componmemnt receives the cell on which it was called in the prop `cursorIndex` (eg. `{ col: 1, row: 3 }`).
You'll have to check for `row === -1` (header cell) or `col === -1` (gutter)

**Reply by ajboni** [reply]
That's perfect! Thanks for all the new additions!!!
